### ISSUE TITLE
Why use label information?

### ISSUE THREAD

**Issue by kli-casia** [buriburisuri/supervised_infogan#1]
As far as I know, GAN is designed for unsupervised learning. Is the use of the label information making GAN meaningless?
Thanks.

**Comment by buriburisuri** [comment]
@kli-nlpr  I agree you.  We can use label info just for researching because we don't know about GAN. 

There are three types of GAN.  1) original, 2) conditioned, 3) labelled.  The latter is easier to train and should be evaluated saparately.

Check this repo https://github.com/buriburisuri/ac-gan.

Thank you.

**Comment by kaihuchen** [comment]
@kli-nlpr @buriburisuri 
It is possible that giving labels in one dimension could help to sharpen the unsupervised learning in other dimensions in the latent representation. Using the faces dataset as an example, if the training samples are tagged with male/female, then perhaps it will help with untangling the difference of hair styles among genders and thus arrives at more precise results. I thought this area warrants some exploration.

**Comment by HencyChen** [comment]
Hi, when I run the train.py file, it shows that line 35 in train.py has error, and the error says:"TypeError: Expected int32, got list containing Tensors of type '_Message' instead."
What should I do?
